lib/matplotlib/axes/_axes.py

```diff
@@ -5871,17 +5878,18 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
                     if np.sum(m) > 0:  # make sure there are counts
                         xmin = np.amin(m[m != 0])
                         # filter out the 0 height bins
-                xmin = max(xmin*0.9, minimum)
+                xmin = max(xmin*0.9, minimum) if not input_empty else minimum
```

Comment:
This discards the `_saved_bounds` information.  I am not strictly sure that this is the best place to do this, but this bit of code is convoluted enough....